proposals/0240-sdl-js-pwa.md

```diff
@@ -40,26 +41,63 @@ The proposed solution is to allow web apps to run in a WebEngine and connect to
 
 > An overview of application runtime environment and how they connect to SDL Core. This proposal adds the WebEngine part.
 
-### Chapter 1: SDL support for WebEngines (foundations)
+### Chapter 1: Web application package
+---------
+
+Running apps on an embedded WebEngine defines a new app platform/runtime. With installing an app on the infotainment system there should be a minimum set of requirements on how the app should be packaged.
+
+1. The app package should be a compressed zip file
+2. It should contain a manifest.json and the sdl.js file
+3. The manifest file should contain:
+   1. a relative path to an html file which is the entry point of the app
+   2. a relative path to an app icon in the app bundle
+   3. SDL app ID
+   4. app name
+   5. optionally per supporting SDL locale:
+      1. app name (overrides global app name) (must be one of the valid app nicknames)
+      2. optionally relative path to an app icon (overrides global app icon)
+      3. optionally TTS name (an array of tts chunks)
+      4. optionally VR app names (an array of strings)
+   6. App version
+   7. Min SDL RPC version supported
+   8. Min SDL Protocol version supported
+4. All script files should be included in the package
+   1. Any `<script>` element with `src` attribute should point to a script file in the package
+   2. No scripts from outside the package should be allowed
+5. The entry point HTML file should refer to the manifest file (`<script src="manifest.json" />`)
+
+#### 1.1 Manifest file
+
+> Note: See appendix for an example manifest file
+
+The manifest should be used for multiple purposes.
+
+The SDL Developer Platform should allow developers to upload app packages which are candidates for app certification. The platform can read the manifest file and automatically read app assets instead of requesting the developer to input the data manually.
```

Comment:
```suggestion
The SDL Developer Portal should allow developers to upload app packages which are candidates for app certification. The platform can read the manifest file and automatically read app assets instead of requesting the developer to input the data manually.
```